umh-core/pkg/service/redpanda_monitor/redpanda_monitor.go

```diff
@@ -0,0 +1,618 @@
+		},
+		Env: map[string]string{},
+		ConfigFiles: map[string]string{
+			"run_redpanda_monitor.sh": scriptContent,
+		},
+	}
+
+	return s6Config, nil
+}
+
+// GetConfig is not implemented, as the config is static
+
+// parseRedpandaLogs parses the logs of a redpanda service and extracts metrics
+func (s *RedpandaMonitorService) parseRedpandaLogs(logs []s6service.LogEntry, tick uint64) (*RedpandaMetricsAndClusterConfig, error) {
+	/*
+		A normal log entry looks like this:
+		START_MARKER
+		Hex encoded gzip data of the metrics
+		MID_MARKER
+		Hex encoded gzip data of the cluster config
+		END_MARKER
+	*/
+
+	if len(logs) == 0 {
+		return nil, fmt.Errorf("no logs provided")
+	}
+
+	// Find the last log entry that contains the END_MARKER
+	endMarkerIndex := -1
+	for i := len(logs) - 1; i >= 0; i-- {
+		if strings.Contains(logs[i].Content, END_MARKER) {
+			endMarkerIndex = i
+			break
+		}
+	}
+
+	if endMarkerIndex == -1 {
+		return nil, fmt.Errorf("no end marker found")
+	}
+
+	// Find the last start marker before the end marker
+	startMarkerIndex := -1
+	for i := endMarkerIndex - 1; i >= 0; i-- {
+		if strings.Contains(logs[i].Content, START_MARKER) {
+			startMarkerIndex = i
+			break
+		}
+	}
+	if startMarkerIndex == -1 {
+		return nil, fmt.Errorf("no start marker found")
+	}
+
+	// Inbetween both, we need to find the mid marker
+	midMarkerIndex := -1
+	for i := startMarkerIndex + 1; i < endMarkerIndex; i++ {
+		if strings.Contains(logs[i].Content, MID_MARKER) {
+			midMarkerIndex = i
+			break
+		}
+	}
+
+	if midMarkerIndex == -1 {
+		return nil, fmt.Errorf("no mid marker found")
+	}
+
+	// We need to extract the lines inbetween the start and mid marker & mid and end marker
+	// Metrics is the first part, cluster config is the second part
+	metricsData := logs[startMarkerIndex+1 : midMarkerIndex]
+	clusterConfigData := logs[midMarkerIndex+1 : endMarkerIndex]
+
+	var metricsDataBytes []byte
+	var clusterConfigDataBytes []byte
+	for _, log := range metricsData {
+		metricsDataBytes = append(metricsDataBytes, log.Content...)
+	}
+	for _, log := range clusterConfigData {
+		clusterConfigDataBytes = append(clusterConfigDataBytes, log.Content...)
+	}
+
+	// Remove the START_MARKER, MID_MARKER and END_MARKER
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(START_MARKER), []byte{})
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(MID_MARKER), []byte{})
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(END_MARKER), []byte{})
+
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(START_MARKER), []byte{})
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(MID_MARKER), []byte{})
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(END_MARKER), []byte{})
+
+	var wg sync.WaitGroup
+	var metrics *RedpandaMetrics
+	var clusterConfig *ClusterConfig
+	var metricsErr, clusterConfigErr error
+
+	wg.Add(2)
```

Comment:
add a comment why you paralellize this. I remember you told me that this can take up a couple of ms?